lib/puppet/type/foreman_hostgroup.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+require 'English'
+Puppet::Type.newtype(:foreman_hostgroup) do
+  desc 'foreman_hostgroup manages hostgroups in foreman.'
+
+  ensurable
+
+  def self.title_patterns
+    [
+      [
+        %r{^(.*)/(.*)$},
+        [
+          [:parent_hostgroup],
+          [:name]
+        ]
+      ],
+      [
+        %r{(.*)},
+        [
+          [:name]
+        ]
+      ]
+    ]
+  end
+  newparam(:name, namevar: true) do
+    desc 'The name of the hostgroup.'
+  end
+
+  newparam(:parent_hostgroup, namevar: true) do
+    desc 'The full title of the parent hostgroup'
+  end
+
+  newparam(:parent_hostgroup_name) do
+    desc 'The name of the parent hostgroup.  This only needs to be given if your hostgroups contain slashes!'
+  end
+
+  newparam(:base_url) do
+    desc 'Foreman\'s base url.'
+  end
+
+  newparam(:effective_user) do
+    desc 'Foreman\'s effective user for the registration (usually admin).'
+  end
+
+  newparam(:consumer_key) do
+    desc 'Foreman oauth consumer_key'
+  end
+
+  newparam(:consumer_secret) do
+    desc 'Foreman oauth consumer_secret'
+  end
+
+  newparam(:ssl_ca) do
+    desc 'Foreman SSL CA (certificate authority) for verification'
+  end
+
+  newparam(:timeout) do
+    desc 'Timeout for HTTP(s) requests'
+
+    munge do |value|
+      value = value.shift if value.is_a?(Array)
+      begin
+        value = Integer(value)
+      rescue ArgumentError
+        raise ArgumentError, 'The timeout must be a number.', $ERROR_INFO.backtrace
+      end
+      [value, 0].max
+    end
+
+    defaultto 500
+  end
```

Comment:
Can these be reused from https://github.com/theforeman/puppet-foreman/blob/master/lib/puppet_x/foreman/common.rb?

Reply:
Yes.  I've had a go at that.  let's see if the tests still pass...